ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt

```diff
@@ -16,18 +17,43 @@ import com.braintreepayments.api.ShopperInsightsAnalytics.VENMO_SELECTED
  */
 class ShopperInsightsClient @VisibleForTesting internal constructor(
     private val paymentReadyAPI: PaymentReadyApi,
-    private val braintreeClient: BraintreeClient
+    private val braintreeClient: BraintreeClient,
+    private val deviceInspector: DeviceInspector
 ) {
+    constructor(braintreeClient: BraintreeClient) : this(
+        PaymentReadyApi(),
+        braintreeClient,
+        DeviceInspector()
+    )
+
     /**
      * Retrieves recommended payment methods based on the provided shopper insights request.
      *
+     * @param context Android context
      * @param request The [ShopperInsightsRequest] containing information about the shopper.
      * @return A [ShopperInsightsResult] object indicating the recommended payment methods.
      */
     fun getRecommendedPaymentMethods(
+        context: Context,
         request: ShopperInsightsRequest,
         callback: ShopperInsightsCallback
     ) {
+        val applicationContext = context.applicationContext
+        val isVenmoAppInstalled = deviceInspector.isVenmoInstalled(applicationContext)
+        val isPayPalAppInstalled = deviceInspector.isPayPalInstalled(applicationContext)
+
+        if (isVenmoAppInstalled && isPayPalAppInstalled) {
```

Comment:
I'm confused why this is an && statement - can you provide more context on why we only want to return both PayPal and Venmo recommended together when both apps are installed, and not one or the other?

Reply:
I'm not sure. That's what the ticket said.
I'll have to ask @scannillo to shed some context there.

Reply:
Ah ok I see baed on the ticket description this looks good 👍

Reply:
Basically, we'll only want to skip the PaymentReady API call if _both_ Venmo & PayPal apps are installed. The logic should be more clear once that API call is added.

[For now on iOS](https://github.com/braintree/braintree_ios/pull/1167/files), I wrote it like this:

```
        if isVenmoAppInstalled() && isPayPalAppInstalled() {
            return BTShopperInsightsResult(isPayPalRecommended: true, isVenmoRecommended: true)
        } else {
            // TODO: - Make API call to PaymentReadyAPI. DTBTSDK-3176
            return BTShopperInsightsResult()
        }
```

There will prob be even further logic after the API call to prefer the `true` value from the app installed check even if the API call returns `false`.